pkg/providers/vsphere/validator.go

```diff
@@ -207,6 +210,19 @@ func (v *Validator) validateControlPlaneIp(ip string) error {
 	return nil
 }
 
+func (v *Validator) validateWorkerNodeGroupName(spec *spec, workerNodeGroupConfigs []anywherev1.WorkerNodeGroupConfiguration) error {
```

Comment:
Interesting, why are we running this only for vsphere? shouldn't be something we apply to all providers?
I was hopping this code to be somewhere in the api package
What do you think?

Reply:
Agreed! Moving it to a more fitting spot